library/src/window_impl.cxx

```diff
@@ -44,6 +55,97 @@ class window_impl::internals
     return this->CachePath;
   }
 
+#if _WIN32
+  /**
+   * Helper function to detect if the
+   * Windows Build Number is equal or greater to a number
+   */
+  BOOL IsWindowsBuildNumberOrGreater(int buildNumber)
+  {
+    int windowsBuildNumber = 0;
+
+    // Receives the handle to Reg key
+    HKEY versionHKey;
+    LONG result = RegOpenKeyExA(HKEY_LOCAL_MACHINE,
+      "SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion", 0, KEY_READ, &versionHKey);
+
+    // Query value from the reg key
+    if (result == ERROR_SUCCESS)
+    {
+      // Receives code indicating the type of data stored in the reg value.
+      // CurrentBuildNumber is of type REG_SZ
+      DWORD type;
+      DWORD dataSize = MAX_VALUE_NAME;
+      CHAR data[MAX_VALUE_NAME]{};
+
+      RegGetValueA(versionHKey, "", "CurrentBuildNumber", RRF_RT_REG_SZ, &type, data, &dataSize);
```

Comment:
Can we use the vtksys wrapper to simplify the code a bit? See https://github.com/Kitware/VTK/blob/697646db54ce4de95ff5adb500799e49feee4935/Utilities/KWSys/vtksys/SystemTools.hxx.in#L844

Reply:
sure